src/SplitClient.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { FunctionComponent, useContext, useMemo } from 'react';
+import { SplitContext } from './SplitProvider';
+import { ISplitContextValues } from './types';
+
+export interface ISplitClientProps {
+  splitKey: SplitIO.SplitKey;
+  trafficType?: string;
+}
+
+export const SplitClient: FunctionComponent<ISplitClientProps> = ({
+  splitKey,
+  trafficType,
+  children,
+}) => {
+  const { factory, isReady, lastUpdate } = useContext(SplitContext);
+
+  // create client for splitKey and trafficType
+  // TODO should this be in an effect with state?
+  const client = useMemo(() => {
+    return factory ? factory.client(splitKey, trafficType) : null;
+  }, [factory, splitKey, trafficType]);
```

Comment:
We might want to change this to `useEffect` with `useState`. Should initializing the client block rendering?...

Reply:
Yeah, since this a call to a 3rd party source, useEffect would make more sense.